src/mimefactory.rs

```diff
@@ -146,6 +156,9 @@ pub struct MimeFactory {
 
     /// This field is used to sustain the topic id of webxdcs needed for peer channels.
     webxdc_topic: Option<TopicId>,
+
+    /// This field is used when this is either a pre-message or a Post-Message.
+    pre_message_mode: Option<PreMessageMode>,
```

Comment:
Why not add e.g. `None` to `PreMessageMode`? This way Option isn't needed, maybe even some code will simplify